### ISSUE TITLE
snippets with {VISUAL}

### ISSUE THREAD

**Issue by iandberg** [garbas/vim-snipmate#233]
Hi,
I'm having issues trying to get the visual mode content in the snippet. In regular visual mode, I select the text, hit <tab>trigger-name<tab>, and I get the literal snippet. The VISUAL bit isn't resolved. For example, using the supplied snippet:

```
snippet is "isset()"
	isset($1{VISUAL})
```
I highlight some text, do tab, is, tab, and I just get

`isset({VISUAL})`

I do have a number of other popular plugs installed, (supertab, commentary, surround, etc). I did try running a stripped down vim without the plugs but had trouble getting Snipmate to work at all.

Are there any known conflicts concerning the VISUAL placeholder? I'm running a recent MacVim and a fresh version of Snipmate. Works great otherwise and I love it.

thanks for any insight!
Ian

**Comment by ajzafar** [comment]
This depends a bit on the version of the snippet parser that you're using (controlled with the `g:snipMate.snippet_version` option, see `:h SnipMate-options`). As mentioned in `:h SnipMate-visual`, in version 1 (the newer version of the parser), both `$VISUAL` and `${VISUAL}` will work, with the `$` being necessary. In version 0 (the legacy parser), *only* `{VISUAL}` is valid, *without* the `$`. It looks like your snippet is written for the latter, but maybe you have SnipMate configured to use the former.

Check for any version lines in your snippet file as well as the output of `:echo g:snipMate.version`. Other plugins shouldn't be interfering.

**Comment by iandberg** [comment]
Thanks - I forgot to mention that I had tried the other 'flavors' of VISUAL as mentioned in the help: $VISUAL, ${VISUAL}  - with the same results (literal output).
When I do `:echo g:snipMate.version`
I get:
```
E716: Key not present in Dictionary: version
E15: Invalid expression: g:snipMate.version
```
I then explicitly set the version in my vimrc
```
let g:snipMate = {}
let g:snipMate.snippet_version = 1
```

And now it works :) The visual variable gets the content now.

Though, `:echo g:snipMate.version` still gives me the dictionary errors, FYI

Thanks for your help!

Ian

**Comment by ajzafar** [comment]
Oh shoot. I meant to say `:echo g:snipMate.snippet_version`. Glad it's working, but it should work even if you don't specify a parser version. I will have to investigate that. Leaving this open till I figure that out.

**Comment by iandberg** [comment]
No worries. Let me know if you can't reproduce - I can try to set the conditions again. Am surprised there's not more chatter about the VISUAL feature - it's very handy.

**Comment by ajzafar** [comment]
It looks like my fix for #177 only worked for VISUAL stops on an indented line. All VISUAL stops should work now. (Both issues only affect the legacy parser). Thanks for the report!
